dashboard/test/controllers/api/v1/peer_review_submissions_controller_test.rb

```diff
@@ -129,6 +129,128 @@ class Api::V1::PeerReviewSubmissionsControllerTest < ActionController::TestCase
+      Timecop.travel 1.day
+      PeerReview.where(level_source: first_answer, status: :escalated).update(
+        reviewer: instructor,
+        from_instructor: true,
+        data: 'Please make some changes',
+        status: :rejected
+      )
+
+      # The learner updates their answer
+      Timecop.travel 1.day
+      second_answer = create :level_source, level: level
+      user_level.update level_source: second_answer
+      # Setup check: Two new peer reviews should exist for this submission
+      assert_equal 2, PeerReview.where(level_source: second_answer).count
+
+      # The instructor reviews again, rejecting the submission a second time
+      Timecop.travel 1.day
+      PeerReview.where(level_source: second_answer, status: :escalated).update(
+        reviewer: instructor,
+        from_instructor: true,
+        data: 'Still not good enough',
+        status: :rejected
+      )
+
+      # The learner updates their answer one more time
+      Timecop.travel 1.day
+      final_answer = create :level_source, level: level
+      user_level.update level_source: final_answer
+      # Setup check: Two new peer reviews should exist for this submission
+      assert_equal 2, PeerReview.where(level_source: final_answer).count
+
+      # Remember this date - we'll check it later
+      last_submission_date = PeerReview.where(level_source: final_answer).last.created_at
+
+      # The instructor reviews a third time, finally accepting the submission
+      Timecop.travel 1.day
+      PeerReview.where(level_source: final_answer, status: :escalated).update(
+        reviewer: instructor,
+        from_instructor: true,
+        data: 'Great work!',
+        status: :accepted
+      )
+
+      # Setup check: We should now see three completed peer reviews and one still
+      # open for a non-instructor.
+      assert_equal 3, PeerReview.where(level: level, submitter: learner, reviewer: instructor).count
+      assert_equal 1, PeerReview.where(level: level, submitter: learner, reviewer: nil).count
+
+      # Jump forward one more day for good measure
+      Timecop.travel 1.day
+
+      # Finally, the thing we wanted to test:
+      # Generate CSV as the instructor
+      sign_in instructor
+      get :report_csv, params: {plc_course_unit_id: course_unit.id}
+      assert_response :success
+      response = CSV.parse(@response.body)
+
+      # Check that the relevant columns are where we expect them to be
+      assert_equal "#{level.name.titleize} First Submit Date", response[0][3]
+      assert_equal "#{level.name.titleize} Latest Submit Date", response[0][4]
+
+      # Check that only our test data was included (a header row and one data row)
+      assert_equal 2, response.count
+
+      # Check columns for correct dates
+      assert_equal first_submission_date.strftime("%-m/%-d/%Y"), response[1][3]
+      assert_equal last_submission_date.strftime("%-m/%-d/%Y"), response[1][4]
+    end
+  end
```

Comment:
It look a long time to figure out how to write this test!  I've tried to write it so that, in some ways, it acts as documentation for how all the peer review stuff fits together.  If you have any questions, please let me know so we can document the answers in the code here!